src/main/java/org/mozilla/jss/CryptoManager.c

```diff
@@ -517,6 +533,251 @@ Java_org_mozilla_jss_CryptoManager_initializeAllNative2
+        goto finish;
+    }
+
+    /*
+     * Initialize the errcode translation table.
+     */
+    JSS_initErrcodeTranslationTable();
+
+    /*
+     * The rest of the initialization (the NSS stuff) is skipped if
+     * the initializeJavaOnly flag is set.
+     */
+    if( initializeJavaOnly) {
+        initialized = PR_TRUE;
+        goto finish;
+    }
+
+
+    /*
+     * Set the PKCS #11 strings
+     */
+    manuChars = JSS_RefJString(env, manuString);
+    libraryChars = JSS_RefJString(env, libraryString);
+    tokChars = JSS_RefJString(env, tokString);
+    keyTokChars = JSS_RefJString(env, keyTokString);
+    slotChars = JSS_RefJString(env, slotString);
+    keySlotChars = JSS_RefJString(env, keySlotString);
+    fipsChars = JSS_RefJString(env, fipsString);
+    fipsKeyChars = JSS_RefJString(env, fipsKeyString);
+    if( (*env)->ExceptionOccurred(env) ) {
+        ASSERT_OUTOFMEM(env);
+        goto finish;
+    }
+    PR_ASSERT( strlen(manuChars) == 33 );
+    PR_ASSERT( strlen(libraryChars) == 33 );
+    PR_ASSERT( strlen(tokChars) == 33 );
+    PR_ASSERT( strlen(keyTokChars) == 33 );
+    PR_ASSERT( strlen(slotChars) == 65 );
+    PR_ASSERT( strlen(keySlotChars) == 65 );
+    PR_ASSERT( strlen(fipsChars) == 65 );
+    PR_ASSERT( strlen(fipsKeyChars) == 65 );
+    PK11_ConfigurePKCS11(   manuChars,
+                            libraryChars,
+                            tokChars,
+                            keyTokChars,
+                            slotChars,
+                            keySlotChars,
+                            fipsChars,
+                            fipsKeyChars,
+                            0, /* minimum pin length */
+                            PR_FALSE /* password required */
+                        );
+
+
+    szConfigDir = JSS_RefJString(env, configDir);
+    /*
+    * Set up arguments to NSS_InitContext
+    */
+    szCertPrefix = JSS_RefJString(env, certPrefix);
+    szKeyPrefix = JSS_RefJString(env, keyPrefix);
+    szSecmodName = JSS_RefJString(env, secmodName);
+
+    initFlags = 0;
+    if( readOnly ) {
+        initFlags |= NSS_INIT_READONLY;
+    }
+    if( noCertDB ) {
+        initFlags |= NSS_INIT_NOCERTDB;
+    }
+    if( noModDB ) {
+        initFlags |= NSS_INIT_NOMODDB;
+    }
+    if( forceOpen ) {
+        initFlags |= NSS_INIT_FORCEOPEN;
+    }
+    if( noRootInit ) {
+        initFlags |= NSS_INIT_NOROOTINIT;
+    }
+    if( optimizeSpace ) {
+        initFlags |= NSS_INIT_OPTIMIZESPACE;
+    }
+    if( PK11ThreadSafe ) {
+        initFlags |= NSS_INIT_PK11THREADSAFE;
+    }
+    if( PK11Reload ) {
+        initFlags |= NSS_INIT_PK11RELOAD;
+    }
+    if( noPK11Finalize ) {
+        initFlags |= NSS_INIT_NOPK11FINALIZE;
+    }
+    if( cooperate ) {
+        initFlags |= NSS_INIT_COOPERATE;
+    }
+
+    /*
+    * Initialize NSS Context.
+    */
+    nssInitContext = NSS_InitContext(szConfigDir, szCertPrefix, szKeyPrefix,
+            szSecmodName,NULL, initFlags);
```

Comment:
looks like in the python-nss code, the second to last parameter depends on the init params
(see https://bugzilla.redhat.com/show_bug.cgi?id=2013674#c5)
                                        py_init_params ? &py_init_params->params : NULL,
Would you also need to provide that or is NULL good enough for your case?

Reply:
This has been resolved by actually providing an argument of NSSInitParams instead of making the PK11 call done before.